In [1]:
from pypot.dynamixel import DxlIO
from pypot import Orbita2D

In [2]:
dxl = DxlIO('/dev/ttyACM0')

In [3]:
ids = dxl.scan()
print('Found {}'.format(ids))

Found [22, 23, 253]


Les ids des moteurs trouvés. le 253 ne compte pas.

Ensuite on met les moteurs en multi-turn (on a besoin de le faire qu'un seule fois)

In [3]:
dxl.set_angle_limit({22:(4095,4095)},convert=False)
dxl.set_angle_limit({23:(4095,4095)},convert=False)

On peut voir les positions des moteurs (en deg)

In [4]:
dxl.get_present_position([22])

(-112.22,)

In [5]:
dxl.get_present_position([23])

(-73.1,)

On crée un Orbita2D avec les ids des 2 moteurs le rapport de réduction et éventuellement la position initiale (on va la changer après).

In [6]:
orb1=Orbita2D.Orbita2D([22,23],[0,0],30.0/26,30.0/26)

On peut allumer/couper les moteurs

In [7]:
Orbita2D.torque_off(dxl,[orb1])

Ensuite on place l'articulation dans sa position zéro et on configure les offsets

In [8]:
orb1.set_offsets(dxl)

On peut allumer les moteurs et tester

In [9]:
Orbita2D.torque_on(dxl,[orb1])

In [13]:
dxl.set_goal_position(orb1.ik([0,0]))

In [11]:
dxl.set_goal_position(orb1.ik([20,20]))

On peut tester un goto smooth, on passe les angles goal (en deg) et la durée du mouvement (en s)

In [14]:
Orbita2D.orb2D_goto(dxl,orb1,[20,20],2)

In [15]:
Orbita2D.orb2D_goto(dxl,orb1,[0,0],2)